README.md

```diff
@@ -68,6 +68,36 @@ logger.log({
 });
 ```
 
+#### Optional color configuration
+If you are using custom log levels with winston, you can provide a map of levels and colors in the transport options:
+
+```javascript
+import winston from 'winston';
+import DiscordTransport from 'winston-discord-transport';
+
+const levelColors = {
+  emerg: 9315107,
```

Comment:
Do `emerg` and some of these other levels exist in winston?

Reply:
These are the `syslog` logging levels as specified by [RFC5424](https://tools.ietf.org/html/rfc5424). winston does support these logging levels if [specified](https://github.com/winstonjs/winston#logging-levels). eg:

```js
const logger = winston.createLogger({
  levels: winston.config.syslog.levels,
  transports: [
    new winston.transports.Console({ level: 'error' })
  ]
});
```

Reply:
I see, so how is it propagated to the transports? 

Should the transport check for both kind of levels? Or does winston propagate the `level` to the transports in the default way (i.e. error, warn, info, etc in that order)?

Reply:
The level is sent to the transport as `info.level`, for example:

```
{
  message: 'An example log',
  level: 'notice',
  service: 'example-service',
  timestamp: '2021-03-29 12:28:30',
  [Symbol(level)]: 'notice',
  [Symbol(message)]: '{"message":"An example log","level":"notice","service":"example-service","timestamp":"2021-03-29 12:28:30"}'
}
```

There doesn't seem to be any way of getting the full log level/color configuration from winston. https://github.com/winstonjs/winston/issues/1425

Perhaps a better approach would be to update the `Colors` interface to use the same type signature as Winston's [`AbstractConfigSetColors`](https://github.com/winstonjs/winston/blob/master/lib/winston/config/index.d.ts#L11):

```ts
interface Colors {
  [key: string]: string | string[];
}
```

This would allow you to configure the transport using the built-in level configurations like this:

```ts
new DiscordTransport({
  webhook: 'https:/your/discord/webhook',
  defaultMeta: { service: 'my_node_service' },
  level: 'notice',
  colors: winston.config.syslog.colors
})
```

Reply:
Nice idea!

Although, I was working on moving this functionality to a standalone discord logger - [https://github.com/sidhantpanda/node-discord-logger](https://github.com/sidhantpanda/node-discord-logger)

And working to use that library inside this transport - [https://github.com/sidhantpanda/winston-discord-transport/blob/feat/use_external_node_discord_lib/src/index.ts](https://github.com/sidhantpanda/winston-discord-transport/blob/feat/use_external_node_discord_lib/src/index.ts) under branch `eat/use_external_node_discord_lib`.

I can create this feature in that library and extend those options here and add you as collaborator in the README. I don't want to discard your efforts here :( 

Or if you feel like it, you can create this pull request there.